core/types/backwards_compat.libevm_test.go

```diff
@@ -302,3 +220,24 @@ func TestBodyRLPCChainCompat(t *testing.T) {
 		})
 	}
 }
+
+// equalHash reports whether `a` and `b` have equal hashes. It allows for nil
+// arguments, returning `true` if both are nil, `false` if only one is nil,
+// otherwise `a.Hash() == b.Hash()`.
+func equalHash[
+	T any, P interface {
+		Hash() common.Hash
+		*T
+	},
+](a, b P) bool {
+	if a == nil && b == nil {
+		return true
+	}
+	if a == nil || b == nil {
+		return false
+	}
+	return a.Hash() == b.Hash()
+}
+
+func (h *Header) equalHash(hh *Header) bool           { return equalHash(h, hh) }
+func (tx *Transaction) equalHash(u *Transaction) bool { return equalHash(tx, u) }
```

Comment:
a bit sketchy to have methods defined in test code, but that's fine too 😄

Reply:
Less common, sure, but I don't think there's anything sketchy about it. A method is just syntactic sugar for `func([receiver], [other args]...)` and we add those helpers in tests all the time.